victor/src/layout.rs

```diff
@@ -172,3 +169,135 @@ fn same_formatting_context_block(
         margin,
     })
 }
+
+impl InlineFormattingContext {
+    fn layout(&self, containing_block: &ContainingBlock) -> (Vec<Fragment>, Length) {
+        self.layout_inline_level_boxes(
+            containing_block,
+            &self.inline_level_boxes,
+            &mut Length::zero(),
+        )
+    }
+
+    fn layout_inline_level_boxes(
+        &self,
+        containing_block: &ContainingBlock,
+        children: &[InlineLevelBox],
+        inline_position: &mut Length,
+    ) -> (Vec<Fragment>, Length) {
+        let mut block_size = Length::zero();
+        let mut fragments = Vec::new();
+        for child in children {
+            let (fragment, child_block_size) =
+                child.layout(containing_block, self, inline_position);
+            fragments.push(fragment);
+            block_size = block_size.max(child_block_size);
+        }
+        (fragments, block_size)
+    }
+}
+
+impl InlineLevelBox {
+    fn layout(
+        &self,
+        containing_block: &ContainingBlock,
+        formatting_context: &InlineFormattingContext,
+        inline_position: &mut Length,
+    ) -> (Fragment, Length) {
+        match self {
+            InlineLevelBox::InlineBox(inline) => {
+                inline.layout(containing_block, formatting_context, inline_position)
+            }
+            InlineLevelBox::TextRun(id) => {
+                formatting_context.text_runs[id.0].layout(inline_position)
```

Comment:
Nit: maybe implement `From<TextRunId>` for `InlineFormattingContext`.

Reply:
You mean `Index`? With one call site I’m not sure it’s worth it, but we can always add that later.